src/NServiceBus.Persistence.AzureTable/SagaPersisters/CompatibilitySettings.cs

```diff
@@ -20,7 +22,10 @@ internal CompatibilitySettings(SettingsHolder settings) : base(settings)
         /// </summary>
         public void DisableSecondaryKeyLookupForSagasCorrelatedByProperties()
         {
+            if (wasAssumeSecondaryIndicesExistSet || wasAssumeSecondaryKeyUsesANonEmptyRowKeySetToThePartitionKeySet)
+                throw new InvalidOperationException("Compatibility mode cannot be disabled when AllowSecondaryKeyLookupToFallbackToFullTableScan or AssumeSecondaryKeyUsesANonEmptyRowKeySetToThePartitionKey are called.");
             this.GetSettings().Set(WellKnownConfigurationKeys.SagaStorageCompatibilityMode, false);
+            wasCompatibilityModeDisabled = true;
```

Comment:
I don't think this is going to work. All the state is in settings and it is perfectly valid for someone to have configuration like

```
persistence.Compatibility().Method();
//somewhere else
persistence.Compatibility().OtherMethod();
```

you need to reach into settings to actually validate the state